niworkflows/reports/core.py

```diff
@@ -386,6 +382,43 @@ def generate_report(self):
         (self.out_dir / self.out_filename).write_text(report_render, encoding='UTF-8')
         return len(self.errors)
 
+    @staticmethod
+    def _process_orderings(orderings, layout):
+        """
+        generate relevant combinations of orderings with observed values
+
+        Returns
+        -------
+
+        entities: list
+            the relevant orderings that had unique values
+        value_combos: list
+            unique value combinations for the entities
+        """
+        # get a set of all unique entity combinations
+        all_value_combos = {tuple(bids_file.get_entities().get(k, None) for k in orderings)
+                            for bids_file in layout.get()}
+        # remove the all None member if it exists
+        none_member = tuple([None for k in orderings])
+        if none_member in all_value_combos:
+            all_value_combos.remove(tuple([None for k in orderings]))
+        # see what values exist for each entity
+        unique_values = [{value[idx] for value in all_value_combos}
+                         for idx in range(len(orderings))]
+        # if all values are None for an entity, we do not want to keep that entity
+        keep_idx = [False if (len(val_set) == 1 and None in val_set) or not val_set
+                    else True for val_set in unique_values]
+        # the "kept" entities
+        entities = list(compress(orderings, keep_idx))
```

Comment:
entities should be the same size as value_combos, right now it is just a single list.

Reply:
nvm, this is correct.